modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,398 @@
+      Nwind.var <- ncdf4::ncvar_def(name = "northward_wind", units = "m s-1", dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = WSpd.var, verbose = verbose)
+      nc <- ncdf4::ncvar_add(nc = nc, v = WDir.var, verbose = verbose)
+      nc <- ncdf4::ncvar_add(nc = nc, v = Ewind.var, verbose = verbose)
+      nc <- ncdf4::ncvar_add(nc = nc, v = Nwind.var, verbose = verbose)
+      ncdata_spd <- neonmet.getVals(dates=WSpdDates,product=availProducts[WSpdLoc[1]],site=site,
+                                datetime=datetime,data_col="windSpeedMean",QF_col="windSpeedFinalQF",
+                                units=c("m s-1", "m s-1"))
+      ncdf4::ncvar_put(nc, varid = WSpd.var, vals = ncdata_spd)
+      ncdata_dir <- neonmet.getVals(dates=WSpdDates,product=availProducts[WSpdLoc[1]],site=site,
+                                datetime=datetime,data_col="windDirMean",QF_col="windDirFinalQF",
+                                units=c("degrees", "degrees"))      
+      
+      ncdf4::ncvar_put(nc, varid = WDir.var, vals = ncdata_dir)     
+      wdir_rad <- udunits2::ud.convert(ncdata_dir,"degrees","radians")
+      ncdata_e <- ncdata_spd * cos(wdir_rad)
+      ncdata_n <- ncdata_spd * sin(wdir_rad)
+      ncdf4::ncvar_put(nc, varid = Ewind.var, vals = ncdata_e)
+      ncdf4::ncvar_put(nc, varid = Nwind.var, vals = ncdata_n)
+    } else {
+      logger.warn("No NEON Wind data")
+    }
+    
+    # NEON.DP1.00041 Soil temp    (take 2cm level which is level 501)
+    soilTLoc = grep("DP1\\.00041",availProducts)
+    soilTDates <- neonmet.getDates(availDates,soilTLoc,startMon,endMon)
+    if (length(soilTDates>0)) {
+      if (verbose) {
+        logger.info("Reading NEON Soil Temp")
+      }
+      soilT.var <- ncdf4::ncvar_def(name = "soil_temperature", 
+                                  units = "K", 
+                                  dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = soilT.var, verbose = verbose)
+      ncdata <- neonmet.getVals(dates=soilTDates,product=availProducts[soilTLoc[1]],site=site,
+                                datetime=datetime,data_col="soilTempMean",
+                                urlstring = "\\.00000\\.001\\.5..\\.(.*)_30_minute",
+                                units=c("celsius", "K"),belowground=TRUE)      
+      ncdf4::ncvar_put(nc, varid = soilT.var, vals = ncdata)
+      browser()
+    } else {
+      logger.warn("No NEON Soil Temp")
+    }
+    
+    # NEON.DP1.00034 CO2 at tower top (alt NEON.DP3.00009 CO2 profile) - not yet avail, don't have variable names
+  ncdf4::nc_close(nc)
+  } #For loop
+  return(invisible(results))
+} #function
+
+neonmet.getDates <- function(availDates,Loc,startMon,endMon) {
+  if (length(Loc)>0) { 
+    Dates <- unlist(availDates[Loc[1]])
+    GoodDates <- which((Dates >= startMon) & (Dates <= endMon))
+    if (length(GoodDates)>0) {
+      return(Dates[GoodDates])
+    } else {
+      return(NULL)
+    }
+  } else { 
+    return(NULL)
```

Comment:
Should these `NULL` returns also come with a warning or an info message? E.g. `Warning: No dates found for parameters X and Y...`